exercises/insertion_sort/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+//insertion sort
+func Sort(s []int) {
+	for i := 1; i < len(s); i++ {
+		x := s[i]
+		j := 0
+		for s[i] > s[j] {
+			j++
+		}
+		for k := i; k > j; k-- {
+			s[k] = s[k-1]
+		}
+		s[j] = x
+	}
+}
+
+//insertion sort
+func Sort2(s []int) {
+	for i := 1; i < len(s); i++ {
+		x := i
+		for j := i; j >= 0; j-- {
```

Comment:
imagine this case: [1 2 3 5 6 7 8 4]. everything is sorted but 4, in it is very obvious, that you don't have to iterate until the beginning of the array, and could stop right after 5 is handled. I would recommend fixing this part - let's not make slow sorting algorithms even slower (-;